### PR TITLE
Support role definitions and assignments via the unified role management endpoints

### PR DESCRIPTION
**Pull request by manicminer**
- Support for [Role Definitions](https://docs.microsoft.com/en-us/graph/api/resources/unifiedroledefinition?view=graph-rest-1.0) via the unified role management endpoint
- Support for [Role Assignments](https://docs.microsoft.com/en-us/graph/api/resources/unifiedroleassignment?view=graph-rest-1.0) via the unified role management endpoint

Test results

Screenshot 2022-01-17 at 15 04 10

### CODE REVIEW

File: msgraph/role_assignments_test.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
+package msgraph_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/manicminer/hamilton/internal/utils"
+
+	"github.com/manicminer/hamilton/internal/test"
```
Comment (jackofallops) [inline msgraph/role_assignments_test.go:9]:
```suggestion
	"github.com/manicminer/hamilton/internal/test"
	"github.com/manicminer/hamilton/internal/utils"
```

File: msgraph/role_definitions_test.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
+package msgraph_test
+
+import (
+	"testing"
+
+	"github.com/manicminer/hamilton/internal/utils"
+
+	"github.com/manicminer/hamilton/internal/test"
```
Comment (jackofallops) [inline msgraph/role_definitions_test.go:8]:
```suggestion
	"github.com/manicminer/hamilton/internal/test"
	"github.com/manicminer/hamilton/internal/utils"
```

**Review by koikonom — APPROVED** [review]
LGTM

**Review by jackofallops — APPROVED** [review]
A couple of super minor suggestions, but otherwise LGTM 👍

**Pull request merged**